lib/blockchain/account.ex

```diff
@@ -428,15 +428,16 @@ defmodule Blockchain.Account do
       %Blockchain.Account{code_hash: <<241, 136, 94, 218, 84, 183, 160, 83, 49, 140, 212, 30,
                                         32, 147, 34, 13, 171, 21, 214, 83, 129, 177, 21, 122, 54, 51, 168,
                                         59, 253, 92, 146, 57>>}
-      iex> MerklePatriciaTree.Trie.get(state, BitHelper.kec(<<1, 2, 3>>))
-      <<1, 2, 3>>
+      iex> MerklePatriciaTree.DB.get(state.db, BitHelper.kec(<<1, 2, 3>>))
+      {:ok, <<1, 2, 3>>}
   """
   @spec put_code(EVM.state, EVM.address, EVM.MachineCode.t) :: EVM.state
   def put_code(state, contract_address, machine_code) do
     kec = BitHelper.kec(machine_code)
 
+    MerklePatriciaTree.DB.put!(state.db, kec, machine_code)
```

Comment:
What's the reason for this change? It just violates the isolation layer that I was hoping to achieve in MerklePatriciaTrie?

Reply:
Agreed, I don't like this either.

From the Yellow paper:

**codeHash**: The hash of the EVM code of this account... All such code fragments are contained in the state database under their corresponding hashes for later retrieval.


This also passes the state test so I think we're doing the "right" thing here.

The problem is that it looks like not all state is stored in the Trie. I think this means we need some way to access the database directly. Maybe we could add a DB interface? I'm not sure where that would live though. Maybe an `exeth_common` repo or something?

Reply:
This is fine. We probably need to do a big refactor later to clean up a lot of these packages. I think we might even want to consolidate a few of these apps into an umbrella, since, for instance, Blockchain, EVM, ExWire etc all function as one unit.

👍 with this.